lib/gui/modules/child-writer.ts

```diff
@@ -302,3 +320,43 @@ ipc.connectTo(IPC_SERVER_ID, () => {
 		ipc.of[IPC_SERVER_ID].emit('ready', {});
 	});
 });
+
+async function saveFileBeforeFlash(
+	imagePath: string,
+	saveUrlImageTo: string,
+	onProgress: (state: ProgressState) => void,
+	onFail: (
+		destination: sdk.sourceDestination.SourceDestination,
+		error: Error,
+	) => void,
+) {
+	const urlImage = new Http({ url: imagePath, avoidRandomAccess: true });
+	const source = await urlImage.getInnerSource();
+	const metadata = await source.getMetadata();
+	const fileName = `${saveUrlImageTo}/${metadata.name}`;
+	let alreadyDownloaded = false;
+	try {
+		alreadyDownloaded = (await fs.stat(fileName)).isFile();
```

Comment:
We can't rely on a file with the same name being present in the downloads folder.
It could be any other file.
It could be a partially downloaded file.

Reply:
> It could be any other file.
> It could be a partially downloaded file.

Would checking the size be accurate enough then? Given that I'll change it so that we save the compressed image instead of the uncompressed one